modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module

```diff
@@ -264,6 +273,30 @@ function manatime_feature_timesheets_add_row_form($form, &$form_state) {
 }
 
 /**
+ * Add a new row in the current timesheet.
+ */
+function manatime_feature_timesheets_add_time_entry_form($form, &$form_state) {
+  $form['duration'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Duration'),
+    '#size' => 60,
+    '#maxlength' => 128,
+    '#required' => TRUE,
+  );
+  $form['description'] = array(
+    '#title' => t('Description'),
+    '#type' => 'textarea',
+    '#maxlength' => 128,
+    '#required' => TRUE,
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Add time entry'),
+  );
+  return $form;
```

Comment:
please use form elements with '#type' => 'value' for all values received in form_state or as function parameters (1/2)